README.md

```diff
@@ -2,31 +2,448 @@
+    assertFalse(StringSearchUtility.containsA(""));
+    assertFalse(StringSearchUtility.containsA("amazing"));
+}
+```
+
+Running the Junit test class `StringSearchUtilityTest` should result in both tests passing.
 
-## Conclusion
 
-A short one or two paragraph summary of the contents of the lessons, recapping
-the learning goals.
+## Testing for characters within an ASCII range 
+
+Let's add another utility method named `containsUppercase`
+to check whether a string contains an uppercase letter 'A' through 'Z'.
+
+We could simply use the logical operator `||` (or) to compare each character in the string
+with each uppercase letter, but that would be an annoying boolean
+expression to have to write:
+
+```java
+public static boolean containsUppercase(String str) {
+    for (char c : str.toCharArray()) {
+        if (c == 'A' || c == 'B' || c == 'C' || ....... || c == 'Z' ) {
+            return true;
+        }
+    }
+    return false;
+}
+```
+
+Notice the decimal values for the uppercase letters in the ASCII table:
+
+![ASCII Table](https://curriculum-content.s3.amazonaws.com/6676/java-mod2-strings/ascii_uppercase.png)
+
+Recall from the previous lesson that we can treat a `char` as a decimal value,
+so 'A' and 65 represent the same character, and 'Z' and 90 represent the same character.
+Thus, we can check if a character is an uppercase letter by testing if
+it falls within an inclusive range of decimal values `65` and `90`,
+or we can simply use 'A' and 'Z' for the range:
+
+
+```java
+public static boolean containsUppercase(String str) {
+    for (char c : str.toCharArray()) {
+        if (c >= 'A' && c <= 'Z') {
+            return true;
+        }
+    }
+    return false;
+}
+```
+
+Let's add a Junit test method:
+
+```java
+@Test
+public void containsUppercase() {
+    assertTrue(StringSearchUtility.containsUppercase("BABY"));
+    assertTrue(StringSearchUtility.containsUppercase("baBy"));
+    assertTrue(StringSearchUtility.containsUppercase("Yes"));
+    assertFalse(StringSearchUtility.containsUppercase("hello!"));
+    assertFalse(StringSearchUtility.containsUppercase(""));
+}
+```
+
+Running the Junit test class `StringSearchUtilityTest` should result in all test methods passing.
+
+
+## Comprehension Check
+
+<details>
+   <summary>Look at the ASCII table. What range of characters would the method count?</summary>
+
+  <p>83 and 90 represent the range 'S' to 'Z'</p>
+
+</details>
+
+
+```java
+public static int countCharsInRange(String str) {
+    int count = 0;
+    for (char c : str.toCharArray()) {
+        if (c >= 83 && c <= 90 ) {
+            count++;
+        }
+    }
+    return count;
+}
+```
+
+
+<details>
+   <summary>Consider the code shown below.
+   The code should print <code>true</code> since the string contains the '<' character,
+   but it prints <code>false</code>.  Can you explain the error?  </summary>
+
+  <p>The for loop should not contain an <code>else</code>.  The value <code>false</code>
```

Comment:
You could even suggest stepping through the code with the debugger and putting in break points to visually show if the student wants/needs to see that visually.

Reply:
ok, updated with debugging suggestion.